src/foam/nanos/notification/SendEmailNotificationDAO.java

```diff
@@ -28,48 +37,84 @@ public SendEmailNotificationDAO(X x, DAO delegate) {
     setX(x);
     setDelegate(delegate);
   }
+
   @Override
   public FObject put_(X x, FObject obj) {
-    DAO userDAO = (DAO) x.get("localUserDAO");
-    AppConfig config     = (AppConfig) x.get("appConfig");
-    EmailService email      = (EmailService) x.get("email");
+    DAO localUserDAO  = (DAO) x.get("localUserDAO");
     Notification notif = (Notification) obj;
-    User user = (User) userDAO.find(notif.getUserId());
     Notification oldNotif = (Notification) getDelegate().find(obj);
 
-    if ( oldNotif != null )
-      return super.put_(x, obj);
+    // Check if the notification should send emails.
+    if ( oldNotif != null || ! notif.getEmailIsEnabled() ) return super.put_(x, obj);
+
+    // A list of user ids we're going to send emails to.
+    Set ids = new HashSet<Long>();
+
+    // If `broadcasted` is set, add all users in the system to the list.
```

Comment:
this logic is already implemented. The bug can be fixed by putting 
`notification.setId(0L);` on SendNotificationDAO.java:51

Reply:
I see, thanks. I didn't know about `SendNotificationDAO`.